dspace/config/crosswalks/oai/metadataFormats/oai_openaire.xsl

```diff
@@ -0,0 +1,1484 @@
+        </datacite:nameIdentifier>
+    </xsl:template>
+
+    <xsl:template match="doc:field[starts-with(@name,'person.identifier.gsid')]" mode="entity_author">
+        <datacite:nameIdentifier nameIdentifierScheme="Google Scholar ID"
+            schemeURI="https://scholar.google.com">
+            <xsl:value-of select="./text()"/>
+        </datacite:nameIdentifier>
+    </xsl:template>
+
+    <xsl:template match="doc:field[starts-with(@name,'person.identifier.isni')]" mode="entity_author">
+        <datacite:nameIdentifier nameIdentifierScheme="ISNI" schemeURI="http://www.isni.org">
+            <xsl:value-of select="./text()"/>
+        </datacite:nameIdentifier>
+    </xsl:template>
+
+    <xsl:template match="doc:field[starts-with(@name,'organization.identifier.*')]" mode="entity_author">
+        <!-- TODO resolve the scheme based on the value -->
+        <datacite:nameIdentifier nameIdentifierScheme="" schemeURI="">
+            <xsl:value-of select="./text()"/>
+        </datacite:nameIdentifier>
+    </xsl:template>
+
+    <xsl:template match="doc:field[starts-with(@name,'organization.identifier.isni')]" mode="entity_author">
+        <datacite:nameIdentifier nameIdentifierScheme="ISNI" schemeURI="http://www.isni.org">
+            <xsl:value-of select="./text()"/>
+        </datacite:nameIdentifier>
+    </xsl:template>
+
+
+    <!-- oaire:fundingReferences -->
+    <!-- https://openaire-guidelines-for-literature-repository-managers.readthedocs.io/en/v4.0.0/field_projectid.html -->
+    <xsl:template match="doc:element[@name='dc']/doc:element[@name='relation']" mode="oaire">
+        <oaire:fundingReferences>
+            <xsl:for-each select="./doc:element/doc:field[@name='value']">
+                <xsl:variable name="isRelatedEntity">
+                    <xsl:call-template name="isRelatedEntity">
+                        <xsl:with-param name="element" select="."/>
+                    </xsl:call-template>
+                </xsl:variable>
+                <!-- if next sibling is authority and starts with virtual:: -->
+                <xsl:if test="$isRelatedEntity = 'true'">
+                    <xsl:variable name="entity">
+                        <xsl:call-template name="buildEntityNode">
+                            <xsl:with-param name="element" select="."/>
+                        </xsl:call-template>
+                    </xsl:variable>
+                    <xsl:apply-templates select="$entity" mode="entity_funding"/>
+                </xsl:if>
+            </xsl:for-each>
+        </oaire:fundingReferences>
+    </xsl:template>
+
+    <!-- oaire:fundingReference -->
+    <xsl:template match="doc:element" mode="entity_funding">
+        <oaire:fundingReference>
+            <xsl:apply-templates select="doc:field" mode="entity_funding"/>
+        </oaire:fundingReference>
+    </xsl:template>
+
+    <xsl:template match="doc:field[starts-with(@name,'dc.relation')]" mode="entity_funding">
```

Comment:
There are about 6 templates here in a row that don't have a comment. I really appreciate you commenting what field each template is dealing with.  Could you comment these 6 too?

Reply:
Just add one description.
https://github.com/DSpace/DSpace/pull/2592/commits/7814a9a1df632f03d9d97191637e722078bba317